services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/extensions/dynamodb/mappingclient/model/DeleteItemEnhancedRequest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2010-2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.extensions.dynamodb.mappingclient.model;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.Expression;
+import software.amazon.awssdk.extensions.dynamodb.mappingclient.Key;
+
+@SdkPublicApi
+public class DeleteItemEnhancedRequest<T> {
+
+    private final Key key;
+    private final Expression conditionExpression;
+
+    private DeleteItemEnhancedRequest(Builder builder) {
+        this.key = builder.key;
+        this.conditionExpression = builder.conditionExpression;
+    }
+
+    public static <T> DeleteItemEnhancedRequest<T> create(Key key) {
```

Comment:
AFAIK, to cover the simplest creation use case and give a shortcut instead of doing builder().build().

Reply:
Is `Key` the "required" field? I'd argue if we add more other fields in the future, this would be obsolete.

I think the consumer builder pattern should reduce the verbosity of builder creation. What do you think?

Reply:
I am on the fence because I like the simplicity of it, but the two way door decision principle does indicate remove (cause we can always put it back). So will remove